packages/components/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,155 @@
+
+  const commonProps = {
+    header: 'Header text',
+    description: 'Description text',
+    hint: 'Hint text',
+    label: 'Label text',
+    testID: 'testCheckbox',
+    onPress: onPressSpy,
+  }
+
+  const errorMsg = 'Error text'
+
+  const getIconName = async () => {
+    const icon = await screen.root.findByType(Icon)
+    return icon.props.name
+  }
+
+  describe('Basic tests', () => {
+    it('initializes correctly', () => {
+      render(<Checkbox {...commonProps} />)
+      expect(screen.getByTestId('testCheckbox')).toBeTruthy()
+    })
+
+    it('renders header, description, hint, icon, and label text', async () => {
+      render(<Checkbox {...commonProps} />)
+      expect(screen.getByText('Header text')).toBeOnTheScreen()
+      expect(screen.getByText('Description text')).toBeOnTheScreen()
+      expect(screen.getByText('Label text')).toBeOnTheScreen()
+      expect(screen.getByText('Hint text')).toBeOnTheScreen()
+      expect(screen.queryByText('(*Required)')).not.toBeOnTheScreen()
+      expect(await getIconName()).toBe('CheckBoxOutlineBlank')
+    })
+
+    it('renders required indicator', async () => {
+      render(<Checkbox {...commonProps} required />)
+      expect(screen.getByText('(*Required)')).toBeOnTheScreen()
+    })
+
+    it('renders checked icon', async () => {
+      render(<Checkbox {...commonProps} checked />)
+      expect(await getIconName()).toBe('CheckBox')
+    })
+
+    it('renders indeterminate icon', async () => {
+      render(<Checkbox {...commonProps} indeterminate />)
+      expect(await getIconName()).toBe('IndeterminateCheckBox')
+    })
+
+    it('overrides checked if indeterminate', async () => {
+      render(<Checkbox {...commonProps} indeterminate checked />)
+      expect(await getIconName()).toBe('IndeterminateCheckBox')
+    })
+
+    it('fires onPress', async () => {
+      render(<Checkbox {...commonProps} />)
+      fireEvent.press(screen.getByRole('checkbox'))
+      expect(onPressSpy).toHaveBeenCalled()
+    })
+  })
+
+  describe('Styling', () => {
+    describe('Light mode', () => {
+      it('tiled variant styling (unchecked)', () => {
+        render(<Checkbox {...commonProps} tile />)
+        expect(screen.getByRole('checkbox')).toHaveStyle({
+          borderWidth: 2,
+          borderRadius: 4,
+          padding: 12,
+          paddingRight: 16,
+          borderColor: '#a9aeb1',
+          backgroundColor: '#ffffff',
+        })
+      })
+
+      it('tiled variant styling (checked)', () => {
+        render(<Checkbox {...commonProps} tile checked />)
+        expect(screen.getByRole('checkbox')).toHaveStyle({
+          borderWidth: 2,
+          borderRadius: 4,
+          padding: 12,
+          paddingRight: 16,
+          borderColor: '#005ea2',
+          backgroundColor: '#ecf1f7',
+        })
+      })
+
+      it('error state', () => {
+        render(<Checkbox {...commonProps} error={errorMsg} />)
+
+        expect(screen.root).toHaveStyle({
```

Comment:
Maybe should also validate the presence of the error text since it doesn't seem other tests do, although that is a bit weird to fall under style tests.